go/pkg/client/tree.go

```diff
@@ -89,6 +155,20 @@ func loadFiles(execRoot string, excl []*command.InputExclusion, path string, fs
 			},
 		}
 		return nil
+	} else if t == command.SymlinkInputType {
+		fs[path] = &fileSysNode{
+			symlink: &symlinkNode{target: meta.Symlink.Target},
+		}
+		if meta.Symlink.IsDangling {
+			return nil
+		}
+		// If we can proceed here, we've already verified that target is
+		// under execRoot, so this should never fail.
+		relTarget, err := getTargetRelPath(execRoot, path, meta.Symlink.Target)
+		if err != nil {
+			panic(fmt.Sprintf("Invalid target=%q err=%v", relTarget, err))
```

Comment:
I used `panic` here instead of returning `err`, because I think this is an internal invariant. If we ever reach here, it means there's a bug in the code.

Reply:
We follow the guildeline of "don't panic" in Go code. In particular, it makes sense here because this is a library, and never want to crash the calling executable. If we have a bug, we should surface an error message that will make it easy to catch and fix it.